### PR TITLE
Added logging state and velocity controls to the login and signup forms

### PR DESCRIPTION
**Pull request by jupiter007**
## Description

We need to add some controls on the frontend to stop users from repeated submit attacks. I implemented a loading state to disable the "submit" button while the POST request is being made. That way a user is prevented from repeatedly pushing the button to start new requests while the request is in progress.

I also added an "attempts" state. After 5 attempts, the user gets an error message telling them that there were too many attempts, and that they need to wait before tying again.

The messaging, and possibly some of the functionality, will be updated when the designs for these pages come in. However, I just wanted to put some minimal defenses in place to remind us to address these with the finalized pages.

Fixes # ([#66](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/66)

## Type of change
Please delete options that are not relevant
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
The associated unit tests were updated to test some basics of the new functionality.

1) Start the app
2) Navigate to either the /login or /signup page
3) Make more than 5 attempts to submit bad data in the form
4) On the sixth try, you should get the "too many attempts" message which should display the amount of minutes the user has to wait to try again.

## Checklist:
- [ ] My code follows the style guidelines of this project (Since the signup and login form pages are just placeholders, and no final style guides are in place, this is not the case.
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I updated the CHANGELOG.md and added documentation if necessary
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have completed manual or automated accessibility testing for my changes (I did not carry out detailed accessibility tests, since these pages are just placeholders until we get a final design in place)
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/login/page.tsx [hunk 1]
```diff
@@ -70,6 +72,14 @@ const LoginPage: React.FC = () => {
 
     const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
+        setLoading(true);
+        setErrors([]); // Clear previous errors
+
+        if (attempts >= 5) {
+            setErrors(['Too many attempts. Please wait before trying again.']);
```
Comment (briri) [inline app/login/page.tsx:79]: If we are setting it to 15 minutes, we should be specific in the message

File: app/signup/page.tsx [hunk 2]
```diff
@@ -36,8 +38,14 @@ const SignUpPage: React.FC = () => {
 
     const handleSignUp = async (event: React.FormEvent) => {
         event.preventDefault();
+        setLoading(true);
         setErrors([]); // Clear previous errors
 
+        if (attempts >= 5) {
+            setErrors(['Too many attempts. Please wait before trying again.']);
```
Comment (briri) [inline app/signup/page.tsx:45]: see comment above

File: app/login/page.tsx [hunk 3]
```diff
@@ -130,7 +153,7 @@ const LoginPage: React.FC = () => {
                     onChange={handleInputChange}
                     required
                 />
-                <button type="submit">Login</button>
+                <button type="submit" disabled={(loading || attempts > 5) ? true : false}>{attempts}{loading ? 'Logging in ...' : 'Login'}</button>
```
Comment (briri) [inline app/login/page.tsx:156]: Looks like you're always showing the number of attempts here. Is that for debug?

**Review by briri — APPROVED** [review]

File: app/login/page.tsx (hunk 1 above)
Comment (jupiter007) [reply app/login/page.tsx:79]: @briri do you think 15 minutes makes sense to you? It was just a random number.

File: app/login/page.tsx (hunk 3 above)
Comment (jupiter007) [reply app/login/page.tsx:156]: Oh thanks for catching that Brian :) Yes, that was for debugging.

File: app/login/page.tsx (hunk 1 above)
Comment (jupiter007) [reply app/login/page.tsx]: @briri I updated the code to use a more dynamic approach to lockout times, making it more difficult for bad actors to commit brute force attacks. I also included the remaining minutes in the message to the user.

Comment (briri) [reply app/login/page.tsx]: Yeah, I think 15 makes sense. Thanks for making it dynamic :)

**Pull request merged**